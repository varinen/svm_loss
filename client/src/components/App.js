import React, {Component} from 'react';
import Header from './Header';
import Demo from './Demo';

class App extends Component {
    render() {
        return (
            <div className="wrapper">
                <Header/>
                <Demo/>
            </div>
        )
    }
}

export default App;
