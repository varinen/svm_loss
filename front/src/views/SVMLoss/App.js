import React from "react";
// nodejs library that concatenates classes
import classNames from "classnames";
// @material-ui/core components
import {makeStyles} from "@material-ui/core/styles";

// @material-ui/icons
import {Typography} from "@material-ui/core";
// core components

import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";

import Parallax from "components/Parallax/Parallax.js";

import styles from "svm_assets/jss/views/demo.js";

import Demo from "./Demo";
import HeaderLinks from "./Header/HeaderLinks.js";
import Header from "./Header/Header.js";
import Footer from "./Demo/Footer";

import DevTools from 'containers/DevTools'

const dashboardRoutes = [];
const useStyles = makeStyles(styles);

export default function App(props) {
    const classes = useStyles();
    const {...rest} = props;
    return (
        <div>
            <Header
                color="transparent"
                routes={dashboardRoutes}
                brand="SVM Loss Demo"
                rightLinks={<HeaderLinks/>}
                fixed
                changeColorOnScroll={{
                    height: 150,
                    color: "white"
                }}
                {...rest}
            />
            <Parallax filter image={require("svm_assets/img/gray.jpg")}>
                <div className={classes.container}>
                    <GridContainer>
                        <GridItem xs={12} sm={12} md={6}>
                            <h1 className={classes.title}>SVM Loss Demo</h1>
                            <Typography variant="body2" className={classes.subtitle} component="p">
                                This projects implements the visualization of
                                optimization loss for a linear multiclass
                                classifier using Python and Flask. The idea is
                                based on the <b>Multiclass Linear Classification
                                Optimization Loss Visualization</b> <a
                                href="http://vision.stanford.edu/teaching/cs231n-demos/linear-classify/"
                                target="_blank" rel="noopener noreferrer">web
                                demo</a> by Justin Johnson. The text below is
                                taken from that demo page as well.
                            </Typography>
                        </GridItem>
                    </GridContainer>
                </div>
            </Parallax>
            <div className={classNames(classes.main, classes.mainRaised)}>
                <div className={classNames(classes.container, classes.pt2)}>
                    <Demo store={props.store}/>
                </div>
            </div>
            <Footer/>
            { (process.env.NODE_ENV !== 'production') && <DevTools /> }
        </div>
    );
}
