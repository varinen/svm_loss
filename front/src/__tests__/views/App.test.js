import React from 'react';
import {shallow} from 'enzyme';
import App from '../../views/SVMLoss/App';

describe('App', () => {
    const component = shallow(<App/>);

    it('renders correctly', () => {
        expect(component).toMatchSnapshot();
    });
});

