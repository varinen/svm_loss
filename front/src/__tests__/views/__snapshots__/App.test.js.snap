// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <Header
    brand="SVM Loss Demo"
    changeColorOnScroll={
      Object {
        "color": "white",
        "height": 150,
      }
    }
    color="transparent"
    fixed={true}
    rightLinks={<HeaderLinks />}
    routes={Array []}
  />
  <Parallax
    filter={true}
    image="gray.png"
  >
    <div
      className="makeStyles-container-14"
    >
      <GridContainer
        className=""
      >
        <GridItem
          className=""
          md={6}
          sm={12}
          xs={12}
        >
          <h1
            className="makeStyles-title-15"
          >
            SVM Loss Demo
          </h1>
          <WithStyles(ForwardRef(Typography))
            className="makeStyles-subtitle-16"
            component="p"
            variant="body2"
          >
            This projects implements the visualization of optimization loss for a linear multiclass classifier using Python and Flask. The idea is based on the 
            <b>
              Multiclass Linear Classification Optimization Loss Visualization
            </b>
             
            <a
              href="http://vision.stanford.edu/teaching/cs231n-demos/linear-classify/"
              rel="noopener noreferrer"
              target="_blank"
            >
              web demo
            </a>
             by Justin Johnson. The text below is taken from that demo page as well.
          </WithStyles(ForwardRef(Typography))>
        </GridItem>
      </GridContainer>
    </div>
  </Parallax>
  <div
    className="makeStyles-main-17 makeStyles-mainRaised-18"
  >
    <div
      className="makeStyles-container-14 makeStyles-pt2-3"
    >
      <Demo />
    </div>
  </div>
  <Footer />
  <DevTools />
</div>
`;
